libfastpm/cosmology.c

```diff
@@ -18,6 +18,32 @@
 double HubbleDistance = 2997.92458; /* Mpc/h */
 double HubbleConstant = 100.0;      /* km/s / Mpc/h */
 
+
+void
+fastpm_cosmology_init(FastPMCosmology * c)
+{
+    /* prepare the interpolation object for FD tables. */
+    if (c->N_ncdm > 0) {
+        FastPMFDInterp * FDinterp = malloc(sizeof(FDinterp[0]));
```

Comment:
How come we're still using malloc for FDinterp?

Reply:
Because we can still use cosmology->FDInterp == NULL to detect if we need to free the member -- however I forgot to do that in fastpm_cosmology_destroy. Just added it.

Reply:
But I thought we wanted to use FDinterp[1] to avoid worrying about memory location? Or is this not a problem anymore because we use cosmology[1], and FDinterp is a member of cosmology?

Reply:
Yes along these lines. I think the more definitive picture is that we now have fastpm_cosmology_init and fastpm_cosmology_destro,y and manage the life cycle of FastPMCosmology together with FastPMSolver. Therefore we have addressed to 'worrying about memory allocation' head on.